File: tests/conformance/sandbox-tools/fetch_no_network.php
<?php

return [
    'description' => 'Tool with network disabled',
    'input' => [],
    'permissions' => ['network' => false],
    'execute' => function ($args, $ctx) {
        if (\ZeroMcp\Sandbox::checkNetworkAccess($ctx->toolName, 'localhost', $ctx->permissions)) {
            return ['blocked' => false];
        }
        return ['blocked' => true];
    },
];

File: tests/conformance/sandbox-tools/fetch_no_network.rb
tool description: "Tool with network disabled",
     input: {},
     permissions: { network: false }

execute do |args, ctx|
  if ZeroMcp::Sandbox.check_network_access(ctx.tool_name, "localhost", ctx.permissions)
    { "blocked" => false }
  else
    { "blocked" => true }
  end
end
